src/components/settings/SettingsAccessTab/tests/SettingsAccessGenerateLinkButton.test.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {
+  screen,
+  render,
+  cleanup,
+  act,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import moment from 'moment';
+
+import ContextProvider from '../../SettingsContext';
+import SettingsAccessGenerateLinkButton from '../SettingsAccessGenerateLinkButton';
+import * as hooks from '../../data/hooks';
+import LmsApiService from '../../../../data/services/LmsApiService';
+
+jest.mock('../../../../data/services/LmsApiService', () => ({
+  __esModule: true,
+  default: {
+    createEnterpriseCustomerLink: jest.fn(),
+  },
+}));
+
+const ENTERPRISE_ID = 'test-enterprise';
+const NET_DAYS_UNTIL_EXPIRATION = 100;
+const renderWithContext = (loadingCustomerAgreement = false) => {
+  jest.spyOn(hooks, 'useCustomerAgreementData').mockImplementation(
+    () => ({
+      customerAgreement: { netDaysUntilExpiration: NET_DAYS_UNTIL_EXPIRATION },
+      loadingCustomerAgreement,
+    }),
+  );
+  return (
+    <ContextProvider enterpriseId={ENTERPRISE_ID}>
+      <SettingsAccessGenerateLinkButton onSuccess={() => {}} />
+    </ContextProvider>
+  );
+};
+
+describe('<SettingsAccessGenerateLinkButton />', () => {
+  afterEach(() => {
+    cleanup();
+    jest.clearAllMocks();
+  });
+
+  it('Displays default state', () => {
+    render(renderWithContext());
+    expect(screen.queryByText('Generate link')).toBeTruthy();
+  });
+
+  it('Displays loading state', () => {
+    render(renderWithContext(true));
+    expect(screen.queryByText('Readying link generation')).toBeTruthy();
+  });
+
+  it('Clicking button calls api', async () => {
```

Comment:
Should this test also assert that `onSuccess` is called appropriately with `toHaveBeenCalledTimes` and/or `toHaveBeenCalledWith`?